### DISCUSSION TITLE
Save APIKey

### DISCUSSION THREAD

**Discussion by ai-samira** [ztjhz/BetterChatGPT#209] [Q&A] [answered]
Hello,
We clone BetterChatGPT and running it in my local server and set API key.
but where we open site on another pc , 'BetterChatGPT' asks us api key again , how can I save it so that we can access it with all browsers and all systems?
thanks

**Comment by ztjhz** [comment, marked as answer]
Hi, you can do that by creating a `.env` file in the root directory. Then, you set the `VITE_OPENAI_API_KEY` (see [.env.example](https://github.com/ztjhz/BetterChatGPT/blob/main/.env.example) for more details). Then rebuild the app and your local server will be preloaded with the API key. However, do note that by doing so, your API key will be exposed to anyone who uses your website. This would not be an issue if it is hosted on your private server.

An alternate solution would be to host a proxy server where the API key is set in that server. Then, you can set the `VITE_CUSTOM_API_ENDPOINT` and `VITE_DEFAULT_API_ENDPOINT` to that proxy server. In this way, your API is not exposed to the users using your website. You can learn how to set up your own proxy server [here](https://github.com/ayaka14732/ChatGPTAPIFree).
